PMM-5003 Updated list agent type names

percona-csalguero opened the pull request:
The list commands now use protobuf agent/services types as string
instead of custom strings to match the output of some other commands
like status.
This way processing the response will be easier since all commands
return the same values for the same fields.

percona-csalguero committed:
PMM-5003 Updated list agent type names

The list commands now use protobuf agent/services types as string
instead of custom strings to match the output of some other commands
like status.
This way processing the response will be easier since all commands
return the same values for the same fields.

AlekSi commented on commands/inventory/list_agents.go:
```diff
@@ -16,9 +16,8 @@
-	"strings"
-
 	"github.com/AlekSi/pointer"
+	"github.com/percona/pmm/api/inventorypb"
```
If you run `dep ensure -v`, you will notice that a number of dependencies greatly increased. That's because this package pulls the whole gRPC/protobuf stack. We should not use it pmm-admin. Instead, we use only HTTP/JSON APIs.

percona-csalguero committed:
PMM-5003 pmm-admin list and status are different

Updated to make list and status to use the same names for status and
agent types. Values are hardcoded to avoid introducing big dependencies
but there are references in the code to show from where they were
extracted.

AlekSi commented on commands/inventory/list_agents.go:
```diff
@@ -75,29 +73,31 @@ func (cmd *listAgentsCommand) Run() (commands.Result, error) {
 		}
 		agents = append(agents, listResultAgent{
-			AgentType: "pmm-agent",
+			AgentType: "PMM_AGENT", // inventorypb.AgentType_PMM_AGENT.String()
```
```
agents.ListAgentsBodyAgentTypePMMAGENT
```
from https://github.com/percona/pmm/pull/474

percona-csalguero committed:
PMM-5003 Updated agent & service types

Updated list and list agents results using the new types introduced in
PR #474

AlekSi committed:
Merge branch 'master' into PMM-5003_list_json_fields

AlekSi reviewed (changes requested):
We should also update commands/inventory/list_nodes.go and commands/inventory/list_services.go.

percona-csalguero committed:
PMM-5003 Added types to list nodes & services

Added types from pmm PR #474 to list_nodes and list_services

percona-csalguero committed:
Merge branch 'PMM-5003_list_json_fields' of percona.github.com:percona/pmm-admin into PMM-5003_list_json_fields

AlekSi committed:
Merge branch 'master' into PMM-5003_list_json_fields

AlekSi committed:
Merge branch 'master' into PMM-5003_list_json_fields

BupycHuk committed:
Merge branch 'master' into PMM-5003_list_json_fields

percona-csalguero committed:
PMM-5003 Updated list commands

Updated list commands to use AgentTypeName, ServiceTypeName and
NodeTypeName functions from the pmm pkg

percona-csalguero committed:
Updated dependencies

percona-csalguero committed:
Merge branch 'PMM-5003_list_json_fields' of percona.github.com:percona/pmm-admin into PMM-5003_list_json_fields

percona-csalguero committed:
PMM-5003 Friendly type names

BupycHuk commented on commands/inventory/list_agents.go:
```diff
@@ -75,100 +76,102 @@ func (cmd *listAgentsCommand) Run() (commands.Result, error) {
-		agents = append(agents, listResultAgent{
-			AgentType: "pmm-agent",
+		agentsList = append(agentsList, listResultAgent{
+			AgentType: inventorypb.AgentTypeName("PMM_AGENT"),
```
Please, use constants from pmm repo

percona-csalguero committed:
PMM-5003 New constant for types

BupycHuk commented on commands/inventory/list_agents.go:
```diff
@@ -19,12 +19,29 @@ import (
 	"github.com/percona/pmm-admin/commands"
 )
 
+const (
```
let's move it to pmm repo and figure out how to not import protobuf and grpc packages

BupycHuk commented on commands/inventory/list_agents.go:
```diff
@@ -75,100 +92,102 @@ func (cmd *listAgentsCommand) Run() (commands.Result, error) {
-		agents = append(agents, listResultAgent{
-			AgentType:  "postgres_exporter",
+		agentsList = append(agentsList, listResultAgent{
+			AgentType:  inventorypb.AgentTypeName(AgentTypePostgresExporter),
```
looks like we don't need to use `AgentTypeName` here.

BupycHuk reviewed (changes requested):
Let's add some tests.

BupycHuk committed:
PMM-5003 Remove unnecessary packages.

BupycHuk committed:
Merge branch 'master' into PMM-5003_list_json_fields

# Conflicts:
#	Gopkg.lock

BupycHuk committed:
PMM-5003 Updated output messages.

BupycHuk commented:
@AlekSi PTAL

AlekSi commented on commands/inventory/list_nodes.go:
```diff
@@ -18,10 +18,18 @@ package inventory
 	"github.com/percona/pmm-admin/commands"
 )
 
+var nodeTypeNames = map[string]string{
```
Shouldn't we remove it?

AlekSi reviewed (changes requested):
And a link to up-to-date FB is a must

BupycHuk committed:
PMM-5003 Remove unnecessary map.

BupycHuk committed:
PMM-5003 Reformating.

BupycHuk commented:
@percona-csalguero PTAL

AlekSi commented on Gopkg.toml:
```diff
@@ -11,7 +11,7 @@ required = [
 [[constraint]]
   name = "github.com/percona/pmm"
-  branch = "PMM-2.0"
+  branch = "PMM-5003-fix-service-type-values"
```
It is already merged.

BupycHuk committed:
PMM-5003 Update pmm repo.

BupycHuk committed:
PMM-5003 Fix comments.
